source/structured_data.cc

```diff
@@ -380,6 +381,29 @@ namespace aspect
             column_names.push_back("column " + Utilities::int_to_string(c,2));
         }
 
+      // Make sure the data file actually has as many columns as we think it has
+      // (either based on the header, or based on what was passed to the reinit function).
+      const unsigned int position = in.tellg();
```

Comment:
Your current code works because streampos is implicitly convertable to integers, but the correct type should be this:
```suggestion
      // (either based on the header, or based on what was passed to the constructor).
      const std::streampos position = in.tellg();
```